From 1eee39716c8ad1d9fa75e03f5aadbb6bce1cc295 Mon Sep 17 00:00:00 2001
From: Petr Nechaev <petr.nechaev@cogentembedded.com>
Date: Thu, 10 Mar 2016 00:40:24 -0500
Subject: [PATCH 2/2] Fix shared lib generation

---
 tools/llvm-shlib/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/llvm-shlib/CMakeLists.txt b/tools/llvm-shlib/CMakeLists.txt
index 1abb379..f5a2476 100644
--- a/tools/llvm-shlib/CMakeLists.txt
+++ b/tools/llvm-shlib/CMakeLists.txt
@@ -46,7 +46,7 @@ set(SOURCES
 
 llvm_map_components_to_libnames(LIB_NAMES ${LLVM_DYLIB_COMPONENTS})
 
-if(NOT DEFINED LLVM_DYLIB_EXPORTED_SYMBOL_FILE)
+if(NOT DEFINED LLVM_DYLIB_EXPORTED_SYMBOL_FILE OR NOT ${LLVM_DYLIB_EXPORTED_SYMBOL_FILE})
 
   if( WIN32 AND NOT CYGWIN )
     message(FATAL_ERROR "Auto-generation not implemented for Win32 without GNU utils. Please specify LLVM_EXPORTED_SYMBOL_FILE.")
-- 
2.5.0

