From 0fe9d8b0d139ad3c5bc959cb4919f062b88b67be Mon Sep 17 00:00:00 2001
From: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
Date: Tue, 17 Feb 2015 15:57:35 +0300
Subject: [PATCH 23/29] Avoid double rendering of the same buffer


Signed-off-by: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
---
 ext/wayland/gstwaylandsink.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index d7b108e..380e41b 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -834,6 +834,12 @@ gst_wayland_sink_render (GstBaseSink * bsink, GstBuffer * buffer)
     gst_buffer_unmap (buffer, &src);
   }
 
+  /* drop double rendering */
+  if (G_UNLIKELY (to_render == sink->last_buffer)) {
+    GST_LOG_OBJECT (sink, "Buffer already being rendered");
+    goto done;
+  }
+
   gst_buffer_replace (&sink->last_buffer, to_render);
   render_last_buffer (sink);
 
-- 
1.7.10.4

