From 8be6f164dcd2f684b3fbd83b95944f494b04a26b Mon Sep 17 00:00:00 2001
From: Petr Nechaev <petr.nechaev@cogentembedded.com>
Date: Thu, 29 Jan 2015 20:15:57 +0300
Subject: [PATCH] Enable IPv6 by default

---
 avahi-daemon/avahi-daemon.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/avahi-daemon/avahi-daemon.conf b/avahi-daemon/avahi-daemon.conf
index c992842..a6286ad 100644
--- a/avahi-daemon/avahi-daemon.conf
+++ b/avahi-daemon/avahi-daemon.conf
@@ -23,7 +23,7 @@
 #domain-name=local
 browse-domains=0pointer.de, zeroconf.org
 use-ipv4=yes
-use-ipv6=no
+use-ipv6=yes
 #allow-interfaces=eth0
 #deny-interfaces=eth1
 #check-response-ttl=no
-- 
1.9.3

