diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index b7fe8d6..314a180 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -2592,6 +2592,9 @@ set_surface_type(struct shell_surface *shsurf)
 	struct weston_surface *pes = shsurf->parent;
 	struct weston_view *pev = get_default_view(pes);
 
+	struct weston_matrix *matrix;
+	float scale;
+
 	reset_surface_type(shsurf);
 
 	shsurf->state = shsurf->next_state;
@@ -2606,6 +2609,15 @@ set_surface_type(struct shell_surface *shsurf)
 						 pev->geometry.x + shsurf->transient.x,
 						 pev->geometry.y + shsurf->transient.y);
 		}
+		/*call scale function*/
+		matrix = &shsurf->fullscreen.transform.matrix;
+		weston_matrix_init(matrix);
+		scale = (float)((shsurf->transient.flags)* 0.5);
+		weston_matrix_scale(matrix, scale, scale, 1);
+		wl_list_remove(&shsurf->fullscreen.transform.link);
+		wl_list_insert(&shsurf->view->geometry.transformation_list,
+			       &shsurf->fullscreen.transform.link);
+
 		weston_view_set_position(shsurf->view,
 					 shsurf->transient.x,
 					 shsurf->transient.y);
