From cf309419e948ac13d7ab5e4ef308ecfc99b4b856 Mon Sep 17 00:00:00 2001
From: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
Date: Tue, 17 Feb 2015 15:58:09 +0300
Subject: [PATCH 24/29] Do not create shm pool if we use kms


Signed-off-by: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
---
 ext/wayland/waylandpool.c |   12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/ext/wayland/waylandpool.c b/ext/wayland/waylandpool.c
index 238c41b..c27721e 100644
--- a/ext/wayland/waylandpool.c
+++ b/ext/wayland/waylandpool.c
@@ -375,6 +375,11 @@ gst_wayland_buffer_pool_start (GstBufferPool * pool)
 
   GST_DEBUG_OBJECT (self, "Initializing wayland buffer pool");
 
+#ifdef HAVE_WAYLAND_KMS
+  if (self->display->drm_fd >= 0)
+    goto done;
+#endif
+
   /* configure */
   size = GST_VIDEO_INFO_SIZE (&self->info) * 15;
 
@@ -408,6 +413,7 @@ gst_wayland_buffer_pool_start (GstBufferPool * pool)
   self->size = size;
   self->used = 0;
 
+done:
   return GST_BUFFER_POOL_CLASS (parent_class)->start (pool);
 }
 
@@ -418,6 +424,11 @@ gst_wayland_buffer_pool_stop (GstBufferPool * pool)
 
   GST_DEBUG_OBJECT (self, "Stopping wayland buffer pool");
 
+#ifdef HAVE_WAYLAND_KMS
+  if (self->display->drm_fd >= 0)
+    goto done;
+#endif
+
   munmap (self->data, self->size);
   wl_shm_pool_destroy (self->wl_pool);
 
@@ -431,6 +442,7 @@ gst_wayland_buffer_pool_stop (GstBufferPool * pool)
   g_hash_table_remove_all (self->buffers_map);
   g_mutex_unlock (&self->buffers_map_mutex);
 
+done:
   return GST_BUFFER_POOL_CLASS (parent_class)->stop (pool);
 }
 
-- 
1.7.10.4

