From 4a3e9a475a6688abc81f336a5089e99b15cb5b14 Mon Sep 17 00:00:00 2001
From: Petr Nechaev <petr.nechaev@cogentembedded.com>
Date: Fri, 28 Aug 2015 16:16:09 +0300
Subject: [PATCH] Fix cmake pathes

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 860a6ee..174f35d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -164,7 +164,7 @@ configure_file(
 # install tree package config
 configure_package_config_file(
     cmake/glmConfig.cmake.in
-    ${GLM_INSTALL_CONFIGDIR}/glmConfig.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/${GLM_INSTALL_CONFIGDIR}/glmConfig.cmake
     INSTALL_DESTINATION ${GLM_INSTALL_CONFIGDIR}
     PATH_VARS CMAKE_INSTALL_INCLUDEDIR
     NO_CHECK_REQUIRED_COMPONENTS_MACRO
-- 
1.9.3

